x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/mapper/SemanticTextFieldMapper.java

```diff
@@ -571,6 +573,12 @@ private String generateInvalidQueryInferenceResultsMessage(StringBuilder baseMes
 
             return baseMessageBuilder.toString();
         }
+
+        @Override
+        public BlockLoader blockLoader(MappedFieldType.BlockLoaderContext blContext) {
+            SourceValueFetcher fetcher = SourceValueFetcher.toString(blContext.sourcePaths(name().concat(".text")));
+            return new BlockSourceReader.BytesRefsBlockLoader(fetcher, BlockSourceReader.lookupMatchingAll());
```

Comment:
Someone else'll have to verify that this is the right way to load these. It's how you load text, after all. But if there's a nicer docvalues way this is stored we should use that if we can.

Reply:
yeah, I think this is somewhat dependent on how `semantic_text` fields store text in ES/Lucene. 

In a ideal world it would be nice if this block loader doesn't have to be changed if/when `semantic_text` fields store text in different way(s) and/or change suffix.

Reply:
Text in `semantic_text` is not stored. It has to be retrieved from source.

Reply:
👍